pkg/operator/operator.go

```diff
@@ -161,15 +180,92 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		syncErrrorMetric.WithLabelValues("SyncError").Set(0)
 	}
 
-	if _, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
-		v1helpers.UpdateConditionFn(availableCnd),
-	); updateErr != nil {
+	updateFuncs := []v1helpers.UpdateStatusFunc{}
+	updateFuncs = append(updateFuncs, v1helpers.UpdateConditionFn(availableCnd))
+	allowUpgradeCond := operatorapi.OperatorCondition{
+		Type:   controllerName + operatorapi.OperatorStatusTypeUpgradeable,
+		Status: operatorapi.ConditionTrue,
+	}
+
+	if blockUpgrade {
+		blockUpgradeMessage := fmt.Sprintf("Marking cluster un-upgradeable because %s", blockUpgradeReason)
+		klog.Infof(blockUpgradeMessage)
+		c.eventRecorder.Warningf("Marking cluster as un-upgradable", "because %s", blockUpgradeReason)
```

Comment:
Event `Reason` should be one word capturing the reason *for the transition into the object's current status* , like `VsphereOlderVersonDetected` or something like that. Or just `DeprecatedVSphere`? "Detected" is sort of implied.

Reply:
`DeprecatedVSphere` sounds like we depracted vsphere support..

Reply:
Sorry, I made typo in VsphereOlderVersonDetected, it should be V**S**phereOlderVers**i**onDetected. BTW, What about DeprecatedVSphereVersion? The version is deprecated and its support will be removed in the next release.